include/vcpkg/dependencies.h

```diff
@@ -40,11 +39,30 @@ namespace vcpkg
         EXCLUDED
     };
 
-    struct InstallPlanAction
+    struct BasicAction
     {
-        static bool compare_by_name(const InstallPlanAction* left, const InstallPlanAction* right);
+        static bool compare_by_name(const BasicAction* left, const BasicAction* right);
```

Comment:
```suggestion
        static bool compare_by_name(const BasicAction& left, const BasicAction& right);
```

?

Reply:
The place this is used doesn't want to sort the true plan (`std::vector<BlahAction>`) -- it instead is building up subsets filtered based on user-relevant factors (`std::vector<BlahAction*> excluded, already_installed, will_install, will_rebuild;`). This is then used to sort those subsets for display to the user.

Would a `std::ref<>`-y thing be appropriate there instead? Would that would allow `compare_by_name()` to expect refs instead of pointers?

Reply:
Ah, so the part that is meaningful is the pointer-ness because you're sorting pointers, not Ts. OK, that makes sense, but then does this need to be a member?

```
struct CompareActionPointerByName {
    bool operator()(const BasicAction* left, const BasicAction* right) const noexcept;
};

// ...

bool CompareActionPointerByName::operator()(const BasicAction* left, const BasicAction* right) const noexcept {
    return left->spec.name() < right->spec.name();
}
```

Seems like it would work in all the places and also avoids a function pointer indirection for every comparison...